src/main/java/multithreading/ExecutorServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package multithreading;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+import org.apache.commons.collections4.ListUtils;
+
+public class ExecutorServiceImpl {
+    private static final int THREAD_POOL_SIZE = Runtime.getRuntime().availableProcessors();
+    private final List<Integer> list;
+
+    public ExecutorServiceImpl(List<Integer> list) {
+        this.list = list;
+    }
+
+    public int execute() {
+        ExecutorService executorService = Executors.newFixedThreadPool(THREAD_POOL_SIZE);
+        List<List<Integer>> listOfSubLists = ListUtils.partition(list,
+                list.size() >= THREAD_POOL_SIZE ? list.size() / THREAD_POOL_SIZE : list.size());
+        List<CalculatorSumCallableImpl> callables = listOfSubLists.stream()
+                .map(CalculatorSumCallableImpl::new)
+                .collect(Collectors.toList());
+        try {
+            List<Future<Integer>> futures = executorService.invokeAll(callables);
+            executorService.shutdown();
```

Comment:
I think it's not required

Reply:
What are your mean?
As for me, it's  enought, because shutdown() is a void method.

Reply:
all good here